app/src/main/java/com/sun/makeupwindow/ui/home/HomeFragment.kt

```diff
@@ -0,0 +1,77 @@
+package com.sun.makeupwindow.ui.home
+
+import com.sun.makeupwindow.R
+import com.sun.makeupwindow.base.BaseFragment
+import com.sun.makeupwindow.data.model.Category
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.ui.dialog.LoadingDialog
+import com.sun.makeupwindow.utlis.RepositoryFactory
+import com.sun.makeupwindow.utlis.showToast
+import kotlinx.android.synthetic.main.fragment_home.*
+
+class HomeFragment : BaseFragment(), HomeContract.View {
+
+    private var presenter: HomePresenter? = null
+    private var loadingDialog: LoadingDialog? = null
+    private val adapterProduct = ProductAdapter(this::itemProductClicked)
```

Comment:
bỏ this đi có sao k

Reply:
hình như là có sao anh ạ, em thấy bảo lỗi